superset-frontend/src/explore/components/controls/AnnotationLayerControl/AnnotationLayer.jsx

```diff
@@ -204,15 +204,30 @@ export default class AnnotationLayer extends React.PureComponent {
     return sources;
   }
 
-  isValidFormula(value, annotationType) {
+  isValidFormula(expression, annotationType) {
     if (annotationType === ANNOTATION_TYPES.FORMULA) {
       try {
-        mathjsParse(value).compile().evaluate({ x: 0 });
+        const token = {
+          type: 3,
+          token: 'x',
+          show: 'x',
+          value: 'x',
+        };
+
+        // handle input like "y = x+1" instead of just "x+1"
+        const subexpressions = expression.split('=');
+        if (
+          subexpressions.length > 2 ||
+          (subexpressions[1] && !subexpressions[0].match(/^[a-zA-Z]\w*$/))
+        ) {
+          return false;
+        }
+        mexp.eval(subexpressions[1] ?? subexpressions[0], [token], { x: 0 });
```

Comment:
do we have unit tests for this change? while i agree it's good to replace mathjs, i would appreciate it if we:
- pulled some real life annotation formula values from the database (if you need examples, i can provide some)
- wrote unit tests based off those
- ensure they still pass after changing the dev

Reply:
@etr2460 I added some in `superset-ui` when I removed `mathjs` from the viz plugins: https://github.com/apache-superset/superset-ui/pull/1362. If you are able to pull in a set of real life test cases from you env I'd be happy to add some more complex test cases to the test suite 👍

Reply:
@villebro @kgabryje: here are some formula examples. most of them are constant numbers (.75, 300, etc.) but there are some more complex ones:
- `2^(x/2000)*100`
- `x=0.999`
- `0.0005708176773515392`
- `.5x+0`
- `y=7.00`
- `20.51 *(x < 1577836800000 ) +  20.2*(x < 1580515200000 & x >= 1577836800000) +  19.9*(x < 1583020800000 & x >= 1580515200000) +  19.3*(x < 1585699200000 & x >= 1583020800000) +  18.7*(x < 1588291200000 & x >= 1585699200000) +  18.1*(x < 1590969600000 & x >= 1588291200000) +  18.0*(x < 1593561600000 & x >= 1590969600000) +  18.0*(x < 1596240000000 & x >= 1593561600000) +  17.9*(x < 1598918400000 & x >= 1596240000000) +  17.9*(x < 1601510400000 & x >= 1598918400000) +  17.9*(x < 1604188800000 & x >= 1601510400000) +  17.9*(x < 1606780800000 & x >= 1604188800000) +  17.9*(x < 1609459200000 & x >= 1606780800000) +  17.9*(                    x >= 1609459200000)`

That last one is a real doozy, and probably is the most important one to test here

Reply:
Thanks for providing these @etr2460 ! It turns out we needed some additional functionality from `mathjs` to provide adequate backward compatibility for these more complex cases. I opened this PR on `superset-ui` to centralize this logic in one place and add proper unit tests: https://github.com/apache-superset/superset-ui/pull/1371